File: first semester/homework 6/6task3/6task3/stack.cpp
#include "stack.h"
#include <iostream>

using namespace std;

struct ListElement
{
	ElementType value;
	ListElement *next;
};

struct Stack
{
    ListElement *head;
};

Stack *create()
{
	Stack *stack = new Stack;
	stack->head = nullptr;
	return stack;
}
 
void push(Stack* &stack, ElementType value)
{
	ListElement *temp = new ListElement;     
	temp->value = value;  
	temp->next = stack->head;      
	stack->head = temp; 	              
}

ElementType pop(Stack* &stack)
{  
	ElementType thisValue = stack->head->value;    
	ListElement *temp = stack->head;      
	stack->head = stack->head->next;     
    delete temp;              
    return thisValue;           
}

void clearStack(Stack* &stack)
{
	while (stack->head != nullptr)
	{
		pop(stack);
	}
	delete stack;
}

File: second semester/Homework7_task1/Homework7_task1/Homework7_task1/Stack.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Homework7_task1
{
    /// <summary>
    /// Stack.
    /// </summary>
    public class Stack <ElementType>
    {
        private class StackElement
        {
            public ElementType Value { get; set; }
            /// <summary>
            /// Constructor for Element of stack.
            /// </summary>
            /// <param name="value">Value to be create in stack.</param>
            public StackElement(ElementType value)
            {
                Value = value;
            }
            /// <summary>
            /// Next element in stack.
            /// </summary>
            public StackElement Next { get; set; }
        }

        private StackElement head = null;

        /// <summary>
        /// Push value to a stack.
        /// </summary>
        /// <param name="value">Value to be pushed.</param>
        public void Push(ElementType value)
        {
            var newElement = new StackElement(value)
            {
                Next = head,
                Value = value
            };
            head = newElement;
        }

        /// <summary>
        /// Get value from stack.
        /// </summary>
        public ElementType Pop()
        {
            if (IsEmpty())
            {
                throw new Exception("Stack is empty.");
            }

            var temp = head.Value;
            head = head.Next;
            return temp;
        }

        /// <summary>
        /// Checking stack. Empty?
        /// </summary>
        /// <returns></returns>
        public bool IsEmpty()
        {
            return head == null;
        }
    }
}
